modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/exec/rel/SortNode.java

```diff
@@ -39,14 +41,33 @@
     /** Rows buffer. */
     private final PriorityQueue<Row> rows;
 
+    /** SQL select limit. Negative if disabled. */
+    private final int limit;
+
     /**
      * @param ctx Execution context.
      * @param comp Rows comparator.
+     * @param offset Offset.
+     * @param limit Limit.
      */
-    public SortNode(ExecutionContext<Row> ctx, RelDataType rowType, Comparator<Row> comp) {
+    public SortNode(ExecutionContext<Row> ctx, RelDataType rowType, Comparator<Row> comp,
+        @Nullable Supplier<Integer> offset, @Nullable Supplier<Integer> limit) {
         super(ctx, rowType);
 
         rows = comp == null ? new PriorityQueue<>() : new PriorityQueue<>(comp);
+
+        assert limit == null || limit.get() >= 0;
```

Comment:
Shouldn't be assert here, since limit and offset depends on user input, if user set wrong values for limit and offset there should be correct exception with correct error message.

Reply:
Removed